src/main/java/org/spongepowered/common/data/processor/common/SpawnerUtils.java

```diff
@@ -0,0 +1,100 @@
+            weight = logic.getRandomEntity().itemWeight;
+        }
+
+        Class<? extends Entity> entityClass = EntityList.stringToClassMapping.getOrDefault(logic.getEntityNameToSpawn(), EntityPig.class);
+        EntityType type = EntityTypeRegistryModule.getInstance().getForClass(entityClass);
+
+        Optional<Entity> nextEntity = createEntity(type, logic.getSpawnerWorld());
+        if(nextEntity.isPresent()) {
+            logic.spawnNewEntity(nextEntity.get(), false);
+            return Optional.of(new WeightedSerializableObject<>(new SpongeEntitySnapshotBuilder().from(nextEntity.get()).build(), weight));
+        }
+
+        return Optional.empty();
+    }
+
+    public static void setNextSnapshot(MobSpawnerBaseLogic logic, WeightedSerializableObject<EntitySnapshot> value) {
+        String name = EntityList.classToStringMapping.get(value.get().getType().getEntityClass());
+        logic.mobID = name;
+        // We can create based on NBT
+        if ((value.get() instanceof SpongeEntitySnapshot) && ((SpongeEntitySnapshot) value.get()).getCompound().isPresent()) {
+            SpongeEntitySnapshot snapshot = (SpongeEntitySnapshot) value.get();
+
+            NBTTagCompound compound = snapshot.getCompound().get();
+
+            logic.setRandomEntity(logic.new WeightedRandomMinecart(compound, name));
+        } else {
+            // No NBT we can save (unless we serialize data?)
+            logic.setRandomEntity(null);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public static WeightedTable<EntitySnapshot> getSnapshots(MobSpawnerBaseLogic logic) {
+        WeightedTable<EntitySnapshot> possibleEntities = new WeightedTable<>();
+        for(MobSpawnerBaseLogic.WeightedRandomMinecart weightedEntity : logic.minecartToSpawn) {
+            Optional<EntityType> weightedType = EntityTypeRegistryModule.getInstance().getById(weightedEntity.entityType);
+            Optional<Entity> entity = SpawnerUtils.createEntity(weightedType.orElse(EntityTypes.PIG), logic.getSpawnerWorld());
+            if(!entity.isPresent()) continue;
+            logic.spawnNewEntity(entity.get(), false);
+            possibleEntities.add(new WeightedSerializableObject<>(new SpongeEntitySnapshotBuilder().from(entity.get()).build(), weightedEntity.itemWeight));
+        }
+
+        return possibleEntities;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static void setSnapshots(MobSpawnerBaseLogic logic, WeightedTable<EntitySnapshot> table) {
+        logic.minecartToSpawn.clear();
+        for (TableEntry<EntitySnapshot> entry : table) {
+            if(!(entry instanceof WeightedObject)) continue;
+            WeightedObject<EntitySnapshot> object = (WeightedObject<EntitySnapshot>) entry;
+
+            // Can't get NBT
+            if(!(object.get() instanceof SpongeEntitySnapshot)) continue;
+
+            String name = EntityList.classToStringMapping.get(object.get().getType().getEntityClass());
```

Comment:
Forge compat will have to be done in the forge side. Modifying the implementation accordingly (basically, just have a higher priority for the processor).